Add `createdon` date field to modUser (cf. #11472)

krismas opened the pull request:
### What does it do ?

Update MODX xPDO model, Lexicon, modUser class and the ExtJS panel to deal with this new column.
### Why is it needed ?

This PR is related to "#11472 - Add date joined to modUserProfile"
### Related issue(s)/PR(s)
- #8461 : User createdon 
- #12119 : User creation date 
- #4055 : RegisteredoOn date 
- #4591 : Add active and added user dates

krismas committed:
Add `createdon` date field to modUser (cf. #11472)

Update MODX xPDO model, Lexicon, modUser class and the ExtJS panel to
deal with this new column.

bezumkin commented:
I really don`t understand why this field not exists by default from first versions of MODX.

+1

sottwell commented:
Just one of those little annoying things that wasn't quite annoying enough reach the 'must-do' threshold.

+1

pixelchutes commented:
:beers: 

+1

Mark-H commented on core/model/modx/moduser.class.php:
```diff
@@ -81,6 +85,7 @@ public function setSudo($sudo) {
      */
     public function save($cacheFlag = false) {
         $isNew = $this->isNew();
+        if ($isNew) $this->set('createdon', time());
```
When dealing with import scripts and such, checks like this prevent you from setting a predefined createdon date. It might be better if this is changed to something like:

```` php
if ($isNew && $this->get('createdon') < 1) {
   $this->set('createdon', time());
}
````

That way if the createdon is already set, it isn't overwritten.

krismas replied:
@Mark-H You are right. From what I see `modresource.class.php` use: 
```
        if ($this->isNew()) {
            if (!$this->get('createdon')) $this->set('createdon', time());
        }
```
What is your preference?

Mark-H commented:
@krismas could you please follow the template from https://github.com/modxcms/revolution/blob/master/CONTRIBUTING.md#submit-a-pull-request when submitting pull requests?

krismas commented:
> @krismas could you please follow the template from

@Mark-H OK, I will try to make a new PR ASAP

krismas committed:
Check current "created" value as suggested by @Mark-H

« When dealing with import scripts and such, checks like this prevent
you from setting a predefined createdon date. »

Mark-H commented:
@theboxer were you planning to do anything here?

theboxer commented:
Merged to 2.x branch
